src/core/src/node.cpp

```diff
@@ -298,12 +298,12 @@ void ov::Node::set_friendly_name(const string& name) {
 
 ov::Node* ov::Node::get_input_node_ptr(size_t index) const {
     OPENVINO_ASSERT(index < m_inputs.size(), idx_txt, index, out_of_range_txt);
-    return m_inputs[index].get_output().get_node().get();
+    return m_inputs[index].m_output->m_node;
```

Comment:
```suggestion
    return m_inputs[index].get_output().get_raw_pointer_node();
```
And update descriptor API
The main issue here is that shared pointer is created?